src/learner.cc

```diff
@@ -572,44 +715,43 @@ class LearnerImpl : public Learner {
     }
 
     // Renew the version.
-    mparam_.major_version = std::get<0>(Version::Self());
-    mparam_.minor_version = std::get<1>(Version::Self());
+    attr->mparam.major_version = std::get<0>(Version::Self());
+    attr->mparam.minor_version = std::get<1>(Version::Self());
 
-    cfg_["num_class"] = common::ToString(mparam_.num_class);
-    cfg_["num_feature"] = common::ToString(mparam_.num_feature);
+    attr->cfg["num_class"] = common::ToString(attr->mparam.num_class);
+    attr->cfg["num_feature"] = common::ToString(attr->mparam.num_feature);
 
-    auto n = tparam_.__DICT__();
-    cfg_.insert(n.cbegin(), n.cend());
+    auto n = attr->tparam.__DICT__();
+    attr->cfg.insert(n.cbegin(), n.cend());
 
     // copy dsplit from config since it will not run again during restore
-    if (tparam_.dsplit == DataSplitMode::kAuto && rabit::IsDistributed()) {
-      tparam_.dsplit = DataSplitMode::kRow;
+    if (attr->tparam.dsplit == DataSplitMode::kAuto && rabit::IsDistributed()) {
+      attr->tparam.dsplit = DataSplitMode::kRow;
     }
-
-    this->Configure();
+    attr->need_configuration = true;
   }
 
   // Save model into binary format.  The code is about to be deprecated by more robust
   // JSON serialization format.  This function is uneffected by
   // `enable_experimental_json_serialization` as user might enable this flag for pickle
   // while still want a binary output.  As we are progressing at replacing the binary
   // format, there's no need to put too much effort on it.
-  void SaveModel(dmlc::Stream* fo) const override {
-    LearnerModelParamLegacy mparam = mparam_;  // make a copy to potentially modify
+  void SaveModel(LearnerAttributes const* attr, dmlc::Stream* fo) const {
+    LearnerModelParamLegacy mparam = attr->mparam;  // make a copy to potentially modify
     std::vector<std::pair<std::string, std::string> > extra_attr;
     mparam.contain_extra_attrs = 1;
 
     {
       std::vector<std::string> saved_params;
       // check if rabit_bootstrap_cache were set to non zero before adding to checkpoint
-      if (cfg_.find("rabit_bootstrap_cache") != cfg_.end() &&
-        (cfg_.find("rabit_bootstrap_cache"))->second != "0") {
+      if (attr->cfg.find("rabit_bootstrap_cache") != attr->cfg.end() &&
```

Comment:
Not sure what `rabit_bootstrap_cache` is

Reply:
It's from JVM package.